opal/mca/btl/tcp/btl_tcp_proc.c

```diff
@@ -344,7 +344,7 @@ static mca_btl_tcp_interface_t** mca_btl_tcp_retrieve_local_interfaces(mca_btl_t
         /* create entry for this kernel index previously not seen */
         if (OPAL_SUCCESS != rc) {
             index = proc_data->num_local_interfaces++;
-            opal_hash_table_set_value_uint32(&proc_data->local_kindex_to_index, kindex, &index);
+            opal_hash_table_set_value_uint32(&proc_data->local_kindex_to_index, kindex, (void**) index);
```

Comment:
Why void** ? The correct way to cast this is to leave the index as uint32_t, and then first casting it to an uintptr_t and then to a void* (```(void*)((uintptr_t)index)```)